src/uct/ib/rc/base/rc_ep.c

```diff
@@ -520,6 +517,9 @@ void uct_rc_txqp_purge_outstanding(uct_rc_iface_t *iface, uct_rc_txqp_t *txqp,
             desc = ucs_derived_of(op, uct_rc_iface_send_desc_t);
             ucs_mpool_put(desc);
         } else {
+            ucs_assert(!(op->flags & UCT_RC_IFACE_SEND_OP_FLAG_IFACE));
```

Comment:
why need this assert

Reply:
we shouldn't set `UCT_RC_IFACE_SEND_OP_STATUS` on IFACE-owned ops, because `UCT_RC_IFACE_SEND_OP_STATUS` isn't removed then

Reply:
but we would anyway fail when returning the iface op to the memory pool, no?

Reply:
but we fail if we use `uct_rc_iface_put_send_op` function which checks the flag

Reply:
here we just call `op->handler`

Reply:
i would expect the handler to release the op - and fail on the assert in uct_rc_iface_get_send_op later

Reply:
ok, removed